### PR TITLE
Remove the deprecated "show to guests only" function

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

**Comment by ausi** [comment]
Should we add a migration that enables the protected checkbox and adds “guests” to the selected groups?

**Comment by leofeyer** [comment]
Good question. @contao/developers WDYT?

**Comment by fritzmg** [comment]
I am in favour for a migration.

**Comment by Toflar** [comment]
Yes, of course! Otherwise you cannot update.

**Review by m-vo — DISMISSED** [review]
Code looks good to me. Otherwise I'm also on #teammigration. :smile:

**Comment by leofeyer** [comment]
Migration added in b52404f63a7083c6f2f3b9a9f17e356a22829edd.

**Review by fritzmg — COMMENTED** [review]
Should we create an `AbstractGuestsMigration` with a `protected static array $tables`? Through `Controller::isVisibleElement` this functionality was available for arbitrary tables.

File: core-bundle/src/Migration/Version500/GuestsMigration.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                LIMIT 1
+            ");
+
+            foreach ($values as $id => $value) {
+                $groups = StringUtil::deserialize($value, true);
+                $groups[] = '-1';
+
+                $data = [
+                    'protected' => '1',
+                    'groups' => serialize($groups),
```
Comment (ausi) [inline core-bundle/src/Migration/Version500/GuestsMigration.php]:
```suggestion
                    '`groups`' => serialize($groups),
```
This needs to be escaped AFAIK.
https://github.com/doctrine/dbal/blob/2deb141194c962c62a19f53edfdca482190eb5b0/src/Connection.php#L694

Comment (fritzmg) [reply core-bundle/src/Migration/Version500/GuestsMigration.php]: Shouldn't we use `$this->connection->quoteIdentifier()` here though? Although we currently have a mixture of directly using the back tick and using `$this->connection->quoteIdentifier()` in the core.

File: core-bundle/src/Migration/Version500/GuestsMigration.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class GuestsMigration extends AbstractMigration
```
Comment (Toflar) [inline core-bundle/src/Migration/Version500/GuestsMigration.php]: @contao/developers this is another one that looks like it should be an abstract that I can use in my extension. Maybe I did not name it `groups` but I think many extensions have the same issue, don't you think?

**Review by ausi — DISMISSED** [review]

**Review by aschempp — CHANGES_REQUESTED** [review]
Please add an abstract migration, this will certainly be needed by other extensions (e.g. Isotope eCommerce).

File: core-bundle/src/Migration/Version500/GuestsMigration.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/Migration/Version500/GuestsMigration.php]: Abstract migration added in e6647793df121d1a855c9921c3cbad086613ed53.

**Review by Toflar — DISMISSED** [review]

**Review by aschempp — DISMISSED** [review]
❤️

File: core-bundle/src/Migration/Version500/AbstractGuestsMigration.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    public function __construct(private Connection $connection)
+    {
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->createSchemaManager();
+
+        foreach ($this->getTables() as $table) {
+            if (!$schemaManager->tablesExist([$table]) || !isset($schemaManager->listTableColumns($table)['guests'])) {
```
Comment (ausi) [inline core-bundle/src/Migration/Version500/AbstractGuestsMigration.php]:
```suggestion
            if (
                !$schemaManager->tablesExist([$table])
                || !isset($schemaManager->listTableColumns($table)['guests'], $schemaManager->listTableColumns($table)['groups']) 
            ) {
```

File: core-bundle/src/Migration/Version500/AbstractGuestsMigration.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+        return false;
+    }
+
+    public function run(): MigrationResult
+    {
+        $schemaManager = $this->connection->createSchemaManager();
+
+        foreach ($this->getTables() as $table) {
+            if (!$schemaManager->tablesExist([$table]) || !isset($schemaManager->listTableColumns($table)['guests'])) {
```
Comment (ausi) [inline core-bundle/src/Migration/Version500/AbstractGuestsMigration.php]:
```suggestion
            if (!$schemaManager->tablesExist([$table]) || !isset($schemaManager->listTableColumns($table)['guests']) || !isset($schemaManager->listTableColumns($table)['groups'])) {
```

File: core-bundle/src/Migration/Version500/AbstractGuestsMigration.php (hunk 3 above)
Comment (leofeyer) [reply core-bundle/src/Migration/Version500/AbstractGuestsMigration.php]: Changed in 1adcfd7706a4e03e32e645548a0286ef675ce075.

File: core-bundle/src/Migration/Version500/AbstractGuestsMigration.php (hunk 4 above)
Comment (leofeyer) [reply core-bundle/src/Migration/Version500/AbstractGuestsMigration.php]: Changed in 1adcfd7706a4e03e32e645548a0286ef675ce075.

**Review by ausi — APPROVED** [review]

**Review by m-vo — APPROVED** [review]

**Pull request merged**